packages/plugin-core/src/commands/CreateDailyJournal.ts

```diff
@@ -21,22 +50,202 @@ export class CreateDailyJournalCommand extends CreateNoteWithTraitCommand {
     this.key = DENDRON_COMMANDS.CREATE_DAILY_JOURNAL_NOTE.key;
   }
 
-  override execute(opts: CommandOpts): Promise<void> {
+  override async execute(opts: CommandOpts): Promise<CreateDailyJournalData> {
     const config = this._extension.getDWorkspace().config;
     const journalConfig = ConfigUtils.getJournal(config);
+    const maybeDailyVault = journalConfig.dailyVault;
+    const vault = maybeDailyVault
+      ? VaultUtils.getVaultByName({
+          vaults: this._extension.getEngine().vaults,
+          vname: maybeDailyVault,
+        })
+      : undefined;
+    let isFirstTime = false;
+    let isSchemaCreated = false;
+    let isTemplateCreated = false;
 
-    if (_.isUndefined(journalConfig.dailyVault)) {
-      return super.execute({ ...opts });
-    } else {
-      const dailyVault = journalConfig.dailyVault;
-      const vault = dailyVault
-        ? VaultUtils.getVaultByName({
-            vaults: this._extension.getEngine().vaults,
-            vname: dailyVault,
-          })
-        : undefined;
-
-      return super.execute({ ...opts, vaultOverride: vault });
+    const metaData = MetadataService.instance().getMeta();
+    // Only create default schema/template if running Daily Journal for first time after 5/31/22
+    if (_.isUndefined(metaData.firstDailyJournalTime)) {
+      isFirstTime = true;
+      MetadataService.instance().setFirstDailyJournalTime();
+      if (
+        !_.isUndefined(metaData.firstInstall) &&
+        metaData.firstInstall > Time.DateTime.fromISO("2022-05-31").toSeconds()
```

Comment:
NOTE: the user could already have a daily journal schema. this only checks if they have a `dendron.daily` schema. instead of checking for the file, can we check if they have a schema that starts with `daily` pattern?

Reply:
Good point. Fixed to check for schema id instead